src/main/java/bio/terra/service/filedata/google/firestore/FireStoreDao.java

```diff
@@ -437,7 +442,7 @@ FireStoreDirectoryEntry computeDirectory(
             for (Map.Entry<String, List<FireStoreDirectoryEntry>> entry : fileRefsByDatasetId.entrySet()) {
                 // Retrieve the file metadata from Firestore
                 final List<FireStoreFile> fireStoreFiles =
-                    fileDao.batchRetrieveFileMetadata(firestore, entry.getKey(), entry.getValue());
+                    fileDao.batchRetrieveFileMetadata(datasetFirestore, entry.getKey(), entry.getValue());
```

Comment:
This is the main change! It was getting passed the snapshot Firestore before.

Reply:
gah, figures